src/models/ghost.c

```diff
@@ -68,9 +68,20 @@ void freeGhosts()
 void spawnGhost(int ghostId)
 {
     ghostList[ghostId].lastRect = ghostList[ghostId].rects[0];
-    ghostList[ghostId].gridPosition = getInitialPositionOfElement(ghostList[ghostId].ghostElement);
-    setMazeElementAt(ghostList[ghostId].gridPosition, ghostList[ghostId].ghostElement);
+    ghostList[ghostId].gridPosition = getMazePositionOfElement(ghostList[ghostId].ghostElement, initialMaze);
     ghostList[ghostId].uiPosition = gridPosToUiPos(ghostList[ghostId].gridPosition);
+
+    // TODO : reset position in entityMaze
+/*
+    if (!arePositionEquals(
+    ghostList[ghostId].gridPosition,
+    getMazePositionOfElement(ghostList[ghostId].ghostElement, &entityMaze)
+    )) {
+        removeMazeElement(ghostList[ghostId].ghostElement, entityMaze);
+        setMazeElementAt(ghostList[ghostId].gridPosition, ghostList[ghostId].ghostElement, entityMaze);
+    }
+*/
```

Comment:
i don't understand, it's wrong to use entityMaze ? it will remove the coins if a ghost is on a position where there is a coin ?? why don't u create an ghostMaze idk ?

Reply:
it's the maze for the ghosts + pacman to avoid ghost interfering with coins

Reply:
Use :
- entityMaze when handling pacman or ghosts positions
- propsMaze when handling coins & fruits
- initialMaze when looking for obstacles